pyfstat/utils/sft.py

```diff
@@ -175,3 +179,163 @@ def get_official_sft_filename(
     # (see SFTfileIO.c in lalpulsar)
     spec.SFTspan = duration
     return lalpulsar.BuildSFTFilenameFromSpec(spec)
+
+
+def get_sft_constraints_from_tstart_duration(tstart, duration, timestamps=None):
+    """
+    Use start and duration to set up a lalpulsar.SFTConstraints
+    object.
+    """
+    SFTConstraint = lalpulsar.SFTConstraints()
+
+    if tstart is None:
+        SFTConstraint.minStartTime = None
+        SFTConstraint.maxStartTime = None
+    elif duration is None:
+        SFTConstraint.maxStartTime = None
+    else:
+        SFTConstraint.minStartTime = lal.LIGOTimeGPS(tstart)
+        SFTConstraint.maxStartTime = SFTConstraint.minStartTime + duration
+
+    SFTConstraint.timestamps = None  # FIXME: not currently supported
+    if timestamps is not None:
+        raise NotImplementedError("Timestamps not yet supported in this function.")
+
+    logger.info(
+        "SFT Constraints: [minStartTime:{}, maxStartTime:{}]".format(
+            SFTConstraint.minStartTime,
+            SFTConstraint.maxStartTime,
+        )
+    )
+
+    return SFTConstraint
+
+
+def plot_spectrogram(
+    sftfilepattern: str,
+    detector: str,
+    savefig: Optional[bool] = False,
+    outdir: Optional[str] = ".",
+    label: Optional[str] = None,
+    quantity: Optional[str] = "power",
+    sqrtSX: Optional[float] = None,
+    fMin: Optional[float] = None,
+    fMax: Optional[float] = None,
+    constraints: Optional[lalpulsar.SFTConstraints] = None,
+    **kwargs,
+) -> matplotlib.axes.Axes:
+    """
+    Compute spectrograms of a set of SFTs.
+    This is useful to produce visualizations of the Doppler modulation of a CW signal.
+
+    Parameters
+    ----------
+    sftfilepattern:
+        Pattern to match SFTs using wildcards (`*?`) and ranges [0-9];
+        multiple patterns can be given separated by colons.
+    detector:
+        Name of the detector of the data that will be plot.
+    savefig:
+        If True, save the figure in `outdir`.
+        If False, return an axis object without saving to disk.
+    outdir:
+        Output folder.
+    label:
+        Output filename.
+    quantity:
+        Magnitude to be plotted.
+        It can be "power" for power, "normpower" for normalized power, "real" for the real part of
+        the SFTs, and "imag" for the imaginary part of the SFTs.
```

Comment:
to be a bit more explicit:
```suggestion
        It can be "power" for SFT power,
        "normpower" for normalized power (`2*power/(Tsft*sqrtSX**2)`),
        "real" for the real part of the SFTs,
        and "imag" for the imaginary part of the SFTs.
```